File: HumanEvalJs/7.js
//// BEGIN - PROMPT
/**
 * Filter an input array of strings only for ones that contain the given substring.
 *
 * Examples:
 *   filterBySubstring([], 'a'); // []
 *   filterBySubstring(['abc', 'bacd', 'cde', 'array'], 'a'); // ['abc', 'bacd', 'array']
 * @param {string[]} strings - Array of input strings.
 * @param {string} substring - Substring to filter by.
 * @returns {string[]} Filtered array of strings containing the substring.
 */
function filterBySubstring(strings, substring) {
//// BEGIN - CANONICAL SOLUTION
    return strings.filter(x => x.includes(substring));
}

//// BEGIN - TEST
const METADATA = {
    author: 'jt',
    dataset: 'test'
};

const assert = require('assert');

function check(candidate) {
    assert.deepStrictEqual(candidate([], 'john'), []);
    assert.deepStrictEqual(candidate(['xxx', 'asd', 'xxy', 'john doe', 'xxxAAA', 'xxx'], 'xxx'), ['xxx', 'xxxAAA', 'xxx']);
    assert.deepStrictEqual(candidate(['xxx', 'asd', 'aaaxxy', 'john doe', 'xxxAAA', 'xxx'], 'xx'), ['xxx', 'aaaxxy', 'xxxAAA', 'xxx']);
    assert.deepStrictEqual(candidate(['grunt', 'trumpet', 'prune', 'gruesome'], 'run'), ['grunt', 'prune']);
}

//// BEGIN - CHECK
check(filterBySubstring);

File: HumanEval/7.py
## BEGIN - PROMPT
from typing import List


def filter_by_substring(strings: List[str], substring: str) -> List[str]:
    """ Filter an input list of strings only for ones that contain given substring
    >>> filter_by_substring([], 'a')
    []
    >>> filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'a')
    ['abc', 'bacd', 'array']
    """

## BEGIN - CANONICAL SOLUTION
    return [x for x in strings if substring in x]

## BEGIN - TEST


METADATA = {
    'author': 'jt',
    'dataset': 'test'
}


def check(candidate):
    assert candidate([], 'john') == []
    assert candidate(['xxx', 'asd', 'xxy', 'john doe', 'xxxAAA', 'xxx'], 'xxx') == ['xxx', 'xxxAAA', 'xxx']
    assert candidate(['xxx', 'asd', 'aaaxxy', 'john doe', 'xxxAAA', 'xxx'], 'xx') == ['xxx', 'aaaxxy', 'xxxAAA', 'xxx']
    assert candidate(['grunt', 'trumpet', 'prune', 'gruesome'], 'run') == ['grunt', 'prune']

## BEGIN - CHECK
check(filter_by_substring)